src/cbmc/cbmc_parse_options.cpp

```diff
@@ -685,9 +701,36 @@ int cbmc_parse_optionst::get_goto_program(
     {
       status() << "Reading GOTO program from file " << eom;
 
-      if(read_object_and_link(*it, symbol_table, goto_functions,
-                              get_message_handler()))
+      if(read_object_and_link(
+        *it,
+        symbol_table,
+        goto_functions,
+        get_message_handler()))
+      {
         return 6;
+      }
+    }
+
+    if(cmdline.isset("function"))
+    {
+      const std::string &function_id=cmdline.get_value("function");
+      const auto &desired_entry_function=
+        symbol_table.symbols.find(function_id);
+
+      if(desired_entry_function!=symbol_table.symbols.end())
+      {
+        languaget *language=get_language_from_mode(
+          desired_entry_function->second.mode);
+        language->regenerate_start_function(
```

Comment:
@kroening Am I right that language should be wrapped in a `unique_ptr` here as that would be a memory leak?